src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -22,10 +22,14 @@ def update
                       fetch_notifications.where(id: params[:notification_ids])
                     end
     # rubocop:disable Rails/SkipsModelValidations
-    unless notifications.update_all('delivered = !delivered')
+    updated_notifications = notifications.update_all('delivered = !delivered')
+    # rubocop:enable Rails/SkipsModelValidations
+
+    if updated_notifications.blank?
```

Comment:
This code is always going to be **false**, since `update_all` returns the number of records as an integer, not a string. So instead of using `blank?`, we should use `zero?`. We can also avoid checking a second time for the number of updated notifications. The code to send the metrics doesn't have to be in a method, it's just called here and it's quite short.

```ruby
    if updated_notifications.zero?
      # flash message code
    else
      # send metrics to RabbitMQ
    end
```